contracts/service-registry/src/state.rs

```diff
@@ -46,10 +51,110 @@ pub const VERIFIERS_PER_CHAIN: IndexedMap<
 
 /// For now, all verifiers have equal weight, regardless of amount bonded
 pub const VERIFIER_WEIGHT: nonempty::Uint128 = nonempty::Uint128::one();
-
-pub const SERVICES: Map<&ServiceName, Service> = Map::new("services");
 pub const VERIFIERS: Map<(&ServiceName, &VerifierAddress), Verifier> = Map::new("verifiers");
 
+const SERVICES: Map<&ServiceName, Service> = Map::new("services");
+const SERVICE_OVERRIDES: Map<(&ServiceName, &ChainName), ServiceParamsOverride> =
+    Map::new("service_overrides");
+
+pub fn service_with_overrides(
+    storage: &dyn Storage,
+    service_name: &ServiceName,
+    chain: &ChainName,
+) -> error_stack::Result<Service, ContractError> {
+    let service = base_service(storage, service_name)?;
+
+    let params_override = SERVICE_OVERRIDES
+        .may_load(storage, (service_name, chain))
+        .change_context(ContractError::StorageError)?;
+
+    match params_override {
+        Some(params_override) => Ok(Service {
+            min_num_verifiers: params_override
+                .min_num_verifiers
+                .unwrap_or(service.min_num_verifiers),
+            max_num_verifiers: params_override
+                .max_num_verifiers
+                .unwrap_or(service.max_num_verifiers),
+            ..service
+        }),
+        None => Ok(service),
+    }
+}
+
+pub fn base_service(
```

Comment:
I'm not sure if anything outside of the state module needs to be concerned with calling explicitly into base or override. Make the ChainName optional and deal with the difference here